tests/unit/helpers.js

```diff
@@ -171,7 +171,8 @@ export function fake<T>(
   var proto = Object.getPrototypeOf(original);
   for (const key of props) {
```

Comment:
This whole implementation looks somewhat flawed.

The first `while (obj) {` loop collects all property names from the whole prototype chain.
Then the `for (const of props) {` loop iterates over these properties, but skips any property that is not part of the original object or its immediate prototype.

If the logic is correct, then it should suffice to just iterate over the entries of the object and its immediate parent, i.e. remove above from `var props = [];` until the end of the first loop, and change this loop to:

```javascript
for (const key of Object.keys(original).concat(Object.keys(proto))) {
```

Or maybe even `for (const [key, definition] of Object.entries(...)) {`

If you follow this suggestion, then there is no need for the `hasOwnProperty` check any more, and you can thus also remove the eslint suppression rule.

Reply:
`Object.keys` only returns enumerable properties, and so it doesn't work as a replacement for `Object.getOwnPropertyNames`.

Nevertheless, we can still avoid to call `hasOwnProperty` (and the eslint suppression) by just collecting the results from calling `Object.getOwnPropertyNames` from the original object (and its prototype if any).

I've just updated the patch accordingly.